src/Data/Maybe.purs

```diff
@@ -216,6 +217,16 @@ instance showMaybe :: Show a => Show (Maybe a) where
   show (Just x) = "(Just " <> show x <> ")"
   show Nothing  = "Nothing"
 
+instance genericMaybe
+  :: Generic (Maybe a) (Sum (Constructor "Nothing" NoArguments)
+                            (Constructor "Just" (Argument a))) where
+  to (Inl _) = Nothing
+  to (Inr (Constructor (Argument a))) = Just a
+
+  from Nothing = Inl (Constructor NoArguments)
+  from (Just a) = Inr (Constructor (Argument a))
+
+
```

Comment:
Can’t this instance be derived by the compiler?

Reply:
Probably.

Reply:
That was one of the issues having it defined in `Generic` actually; it couldn't derive it there :smile: